src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -1038,11 +1041,7 @@ protected void GenerateFindFirstChar()
                         Ldloc(_runtextposLocal);
                         Ldthisfld(s_runtextstartField);
                         Ble(l1);
-                        Ldthis();
```

Comment:
Nit: The code for `Start` is the same for `Beginning`, sans the field that's loaded. Maybe factor into another function?

Reply:
I'm going to leave this as is right now.